### PR TITLE
Adding code coverage reports. [WIP]

### PR DESCRIPTION
**Pull request by antip00**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -58,10 +59,12 @@ services:
       - ./docker/grafana/provisioning/:/etc/grafana/provisioning/
       - ./docker/grafana/dashboards/:/var/lib/grafana/dashboards/
       - ./docker/grafana/alerting/:/etc/grafana/provisioning/alerting/
+      - ./cover/raw/:/usr/share/grafana/cover
```
Comment (Slach) [inline docker-compose.yaml:62]: please use `coverage` not `cover`, this is more mental name

File: docker-compose.yaml [hunk 2]
```diff
@@ -58,10 +59,12 @@ services:
...
 
     environment:
       GF_INSTALL_PLUGINS: grafana-piechart-panel,grafana-worldmap-panel,grafana-clickhouse-datasource
       GF_LOG_LEVEL: debug
+      GOCOVERDIR: "/usr/share/grafana/cover"
```
Comment (Slach) [inline docker-compose.yaml:67]: please use `coverage` not `cover`, this is more mental name

File: docker-compose.yaml [hunk 3]
```diff
@@ -85,10 +89,12 @@ services:
       - ./docker/grafana/provisioning/:/etc/grafana/provisioning/
       - ./docker/grafana/dashboards/:/var/lib/grafana/dashboards/
       - ./docker/grafana/alerting/:/etc/grafana/provisioning/alerting/
+      - ./cover/raw/:/usr/share/grafana/cover
```
Comment (Slach) [inline docker-compose.yaml:92]: `coverage` not `cover`, this is more mental name

File: docker-compose.yaml [hunk 4]
```diff
@@ -85,10 +89,12 @@ services:
...
 
     environment:
       GF_INSTALL_PLUGINS: grafana-piechart-panel,grafana-worldmap-panel,grafana-clickhouse-datasource
       GF_LOG_LEVEL: debug
+      GOCOVERDIR: "/usr/share/grafana/cover"
```
Comment (Slach) [inline docker-compose.yaml:97]: `coverage` not `cover`, this is more mental name

File: docker-compose.yaml [hunk 5]
```diff
@@ -159,6 +166,68 @@ services:
         chmod +x ./dist/altinity-clickhouse-plugin*
       "
 
+  frontend_coverage_builder:
+    image: node:lts
+    working_dir: /grafana-clickhouse
+    volumes:
+      - ./:/grafana-clickhouse
+      # for speedup npm install under WSL
+      # - "node_modules:/grafana-clickhouse/node_modules"
```
Comment (Slach) [inline docker-compose.yaml:175]: uncomment it

File: docker-compose.yaml [hunk 6]
```diff
@@ -159,6 +166,68 @@ services:
...
+        npm run build:test
+      '
+
+  frontend_coverage_generate:
+    image: node:lts
+    working_dir: /grafana-clickhouse
+    volumes:
+      - ./:/grafana-clickhouse
+      # for speedup npm install under WSL
+      # - "node_modules:/grafana-clickhouse/node_modules"
```
Comment (Slach) [inline docker-compose.yaml:191]: uncomment it

File: docker-compose.yaml [hunk 7]
```diff
@@ -159,6 +166,68 @@ services:
...
+      "
+
+  backend_coverage_generate:
+    image: golang:1.23
+    working_dir: /go/src/grafana-clickhouse
+    volumes:
+      - ./:/go/src/grafana-clickhouse
+    command: |
+      bash -xec "
+        go tool covdata textfmt -i=cover/raw -o=cover/cover.txt &&
```
Comment (Slach) [inline docker-compose.yaml:226]: `coverage/raw` `coverrage/coverage.txt`

File: docker-compose.yaml [hunk 8]
```diff
@@ -159,6 +166,68 @@ services:
...
+        go tool cover -html=cover/cover.txt -o=cover/cover.html
```
Comment (Slach) [inline docker-compose.yaml:227]: coverage/coverage.txt coverrage/coverage.html

**Review by Slach — CHANGES_REQUESTED** [review]

File: .nycrc [hunk 9]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (Slach) [inline .nycrc:1]: is this javascript or JSON?
could we add `/* this is file for create instrumented version of module.js to allow get coverage */`

File: webpack.config.ts [hunk 10]
```diff
@@ -1,24 +1,137 @@
+import CopyWebpackPlugin from 'copy-webpack-plugin';
```
Comment (Slach) [inline webpack.config.ts:1]: if this config using only for instrumented
then let's rename i webpack.config.instrumented.js

File: tsconfig_tests.json [hunk 11]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (Slach) [inline tsconfig_tests.json:1]: rename file to tsconfig.instrumented.js

**Review by Slach — CHANGES_REQUESTED** [review]

File: .nycrc (hunk 9 above)
Comment (antip00) [reply .nycrc:1]: JSON, I can add comment as part of json: `"_comment": "this is file for create instrumented version of module.js to allow get coverage",`. Shall I?

Comment (Slach) [reply .nycrc:1]: yep, please add it

**Review by Slach — APPROVED** [review]

**Pull request merged**